### PR TITLE
refactor: refactor writer

### PR DESCRIPTION
**Pull request by ZENOTME**
This PR
1. clear dead code
2. rename task writer to append only writer
3. separate location generator out to append builder

I find `#[allow(dead_code)] in lib will let us miss lots of dead code in the whole codebase. So I moved it to the type crate and catalog crate.

### CODE REVIEW

File: icelake/src/catalog/mod.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //! This module defines catalog api for icelake.
 
+#![allow(dead_code)]
```
Comment (liurenjie1024) [inline icelake/src/catalog/mod.rs:3]: Why we need this?

File: icelake/src/io/file_writer/equality_delta_writer.rs [hunk 2]
```diff
@@ -46,30 +46,28 @@ pub struct DeltaWriterResult {
...
     inserted_rows: HashMap<OwnedRow, PathOffset>,
     row_converter: RowConverter,
     unique_column_idx: Vec<usize>,
-    file_appender_factory: FileAppenderBuilder<L>,
 }
 
 pub struct EqDeltaWriterMetrics {
-    buffer_path_offset_count: usize,
-    sorted_pos_delete_cache_count: usize,
+    _buffer_path_offset_count: usize,
```
Comment (liurenjie1024) [inline icelake/src/io/file_writer/equality_delta_writer.rs:57]: Is this typo?

File: icelake/src/types/mod.rs [hunk 3]
```diff
@@ -1,6 +1,8 @@
 //! Types will provide the definition of iceberg in-memory data types and
 //! functions to parse from on-disk files.
 
+#![allow(dead_code)]
```
Comment (liurenjie1024) [inline icelake/src/types/mod.rs:4]: Why this?

File: icelake/src/lib.rs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 // Make sure all our public APIs have docs.
 // #![deny(missing_docs)]
-#![allow(dead_code)]
+// #![allow(dead_code)]
```
Comment (liurenjie1024) [inline icelake/src/lib.rs:6]: Just remove it?

File: icelake/src/types/mod.rs (hunk 3 above)
Comment (ZENOTME) [reply icelake/src/types/mod.rs]: Seems there are deprecate fields but we don't use it now. Let's remove it.

File: icelake/src/catalog/mod.rs (hunk 1 above)
Comment (ZENOTME) [reply icelake/src/catalog/mod.rs]: I have removed it now.

**Comment by ZENOTME** [comment]
Maybe we can add typos check?

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**